src/game/Coordinate.java

```diff
@@ -34,23 +34,61 @@ public int getX() {
 	public int getY() {
 		return this.y;
 	}
-	
+
 	public boolean isValid() {
 		return this.valid;
 	}
 
+	public Coordinate down() {
+		return new Coordinate(x, y + 1);
+	}
+	
+	public Coordinate up() {
+		return new Coordinate(x, y - 1);
+	}
+	
+	public Coordinate left() {
+		return new Coordinate(x - 1, y);
+	}
+	
+	public Coordinate right() {
+		return new Coordinate(x + 1, y);
+	}
+	
+	/**
+	 * @return Returns a new Coordinate object that is equivalent to the caller
+	 *         object.
+	 */
+	public Coordinate clone() {
+		return new Coordinate(this.x, this.y);
+	}
+
+	/**
+	 * 
+	 * @param other
+	 *            Coordinate to be compared to
+	 * @return Returns true if both the caller Coordinate and parameter
+	 *         Coordinate are within one position of each other relative to a
+	 *         single board tile in either the X or Y direction. Returns false
+	 *         if the two objects are equivalent.
+	 */
```

Comment:
you might want to break this up into horizontal adjacent and vertical adjacent, to make it a bit easier on the eyes